sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaKeyLoaderSchedulerTest.java

```diff
@@ -0,0 +1,110 @@
+      RsaKeyLoaderScheduler.create(keyLoader, executorService);
+  private boolean elseCondition = false;
+
+  @Test
+  public void shouldNotBeAuthenticatedWithRevokedKey() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+    final RSAPublicKey keyToBeRevoked = Mockito.mock(RSAPublicKey.class);
+    final String kidForKeyToBeRevoked = "kidForKeyToBeRevoked";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return Arrays.asList(
+                  new LoadedPublicKey(kidForValidKey, validKey, this),
+                  new LoadedPublicKey(kidForKeyToBeRevoked, keyToBeRevoked, this));
+            } else {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.internalStart(0, RELOAD_PERIOD, TimeUnit.SECONDS);
+
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isSameAs(keyToBeRevoked);
+            });
+
+    // when -> simulate revoking of key:
+    elseCondition = true;
+
+    // then
+    await()
+        .pollDelay(RELOAD_PERIOD + 1, TimeUnit.SECONDS)
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              // revoked key not found anymore
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isNull();
+            });
+  }
+
+  @Test
+  public void shouldContinueJobProperlyAfterError() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            } else {
+              throw new KeyLoadFailedException("Some error occurred during reload");
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.internalStart(0, RELOAD_PERIOD, TimeUnit.SECONDS);
+    await().untilAsserted(() -> assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey));
+
+    // when -> simulate error-occurrence:
+    elseCondition = true;
+
+    // then:
+    await()
+        // wait 1 second longer then reload-period to make sure the reload-job is executed before
+        // checking:
+        .pollDelay(RELOAD_PERIOD + 1, TimeUnit.SECONDS)
```

Comment:
I think it is better to count the invocations and wait until another invocation happened rather than adding another second to the test runtime.

Reply:
I gave it a new try - hopefully it's more concise now.